src/details/ArborX_DetailsFDBSCANDenseBox.hpp

```diff
@@ -51,6 +51,26 @@ struct CartesianGrid
     size_t constexpr max_size_t = std::numeric_limits<size_t>::max();
     ARBORX_ASSERT(_nx == 0 || _ny == 0 || _nz == 0 ||
                   (_ny < max_size_t / _nx && _nz < max_size_t / (_nx * _ny)));
+
+    // Catch a potential loss of precision that may happen in cellBox() and can
+    // lead to wrong results. Only FDBSCAN-DenseBox suffers from this issue.
+    // FDBSCAN uses only min/max to construct bounding volumes, and thus has no
+    // loss.
+    //
+    // The machine precision by itself is not sufficient. In some experiments
+    // run with a full NGSIM datasets, values below 3 could still produce wrong
+    // results. This may still not be conservative enough, but all runs passed
+    // verification when this warning was not triggered.
+    float constexpr eps = 5 * std::numeric_limits<float>::epsilon();
+    if (std::abs(_h / min_corner[0]) < eps ||
+        std::abs(_h / min_corner[1]) < eps ||
+        std::abs(_h / min_corner[2]) < eps ||
+        std::abs(_h / max_corner[0]) < eps ||
+        std::abs(_h / max_corner[1]) < eps ||
+        std::abs(_h / max_corner[2]) < eps)
+      printf("ArborX warning: FDBSCAN-DenseBox algorithm may experience loss "
+             "of precision, producing wrong results. Please consider switching "
+             "to FDBSCAN.\n");
```

Comment:
Any good reason to prefer `printf` to the C++ I/O facility.  Also consider printing to the standard error stream.

Reply:
No good reason. Would like to change?

Reply:
I switched to `std::cerr`.